### PR TITLE
Adding GCMR balance of plant model

### PR DESCRIPTION
**Pull request by licharlot**
Refs #278

### CODE REVIEW

**Review by GiudGiud — COMMENTED** [review]
made some edits directly and added some comments

**Review by GiudGiud — APPROVED** [review]
looks good. Thanks for taking care of testing too!

**Comment by licharlot** [comment]
Did I put the right executable pattern in the testing? It looks like it is skipped.

File: microreactors/gcmr/balance_of_plant/tests [hunk 1]
```diff
@@ -0,0 +1,23 @@
+[Tests]
+  [startup_syntax]
+    type = 'RunApp'
+    input = 'htgr_startup_transient.i'
+    check_input = true
+    executable_pattern = 'moose*'
```
Comment (GiudGiud) [inline microreactors/gcmr/balance_of_plant/tests:6]:
```suggestion
    executable_pattern = 'moose*|relap-7*'
```

File: microreactors/gcmr/balance_of_plant/tests [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  []
+  [load_follow_syntax]
+    type = 'RunApp'
+    input = 'htgr_load_follow_transient.i'
+    check_input = true
+    executable_pattern = 'moose*'
```
Comment (GiudGiud) [inline microreactors/gcmr/balance_of_plant/tests:12]:
```suggestion
    executable_pattern = 'moose*|relap-7*'
```

File: microreactors/gcmr/balance_of_plant/tests [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  []
+  [startup]
+    type = 'CSVDiff'
+    input = 'htgr_startup_transient.i'
+    csvdiff = 'htgr_startup_transient.csv'
+    cli_args = "Executioner/num_steps=10"
+    rel_err = 1e-6
+    abs_zero = 1e-7
+    executable_pattern = 'moose*|thermal_hydraulics*'
```
Comment (GiudGiud) [inline microreactors/gcmr/balance_of_plant/tests:21]:
```suggestion
    executable_pattern = 'moose*|thermal_hydraulics*|relap-7*'
```

**Comment by GiudGiud** [comment]
good catch

**Comment by licharlot** [comment]
The RELAP-7 executable used for testing is too old and doesn't have the new closures set.  Not sure how to change it to use a newer version. (the HPC binary works)

**Comment by GiudGiud** [comment]
new submodule should do the trick

**Comment by GiudGiud** [comment]
ok we ll just wait for relap 7 to catch up with MOOSE if that's ok with you

**Comment by GiudGiud** [comment]
I ll fixup the test

**Comment by GiudGiud** [comment]
Passing now. Thanks for double checking

**Pull request merged**